e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -221,6 +247,39 @@ public async Task Method_ServiceInvokeDeviceMethodWithUnknownModuleThrows()
             await serviceClient.CloseAsync().ConfigureAwait(false);
         }
 
+        public static async Task ServiceSendMethodAndVerifyNotReceivedAsync(string deviceId, string methodName, MsTestLogger logger, TimeSpan responseTimeout = default, ServiceClientTransportSettings serviceClientTransportSettings = default)
+        {
+            ServiceClient serviceClient = null;
+            if (serviceClientTransportSettings == default)
+            {
+                serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            }
+            else
+            {
+                serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString, TransportType.Amqp, serviceClientTransportSettings);
+            }
+
+            TimeSpan methodTimeout = responseTimeout == default ? s_defaultMethodTimeoutMinutes : responseTimeout;
+            logger.Trace($"{nameof(ServiceSendMethodAndVerifyResponseAsync)}: Invoke method {methodName}.");
+            try
+            {
+                CloudToDeviceMethodResult response =
+                    await serviceClient.InvokeDeviceMethodAsync(
+                        deviceId,
+                        new CloudToDeviceMethod(methodName, methodTimeout).SetPayloadJson(null)).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (!(ex is DeviceNotFoundException))
```

Comment:
this should always mandatorily throw an exception correct, so if response is returned then that would be a "fail" case.

you can instead do:
```suggestion
               await Assert.ThrowsExceptionAsync<DeviceNotFoundException>(async () =>
            {
                await serviceClient.InvokeDeviceMethodAsync(
                        deviceId,
                        new CloudToDeviceMethod(methodName, methodTimeout).SetPayloadJson(null)).ConfigureAwait(false);
            });
```

Reply:
I did not get it. We can discuss offline

Reply:
sure, give me call when you want to discuss this